packages/app/src/systems/Chains/eth/containers/EthAssetsDialog.tsx

```diff
@@ -1,40 +1,180 @@
-          ))}
+        <Box.Stack css={{ pb: '$2' }}>
+          <Controller
+            name="address"
+            control={form.control}
+            render={(props) => {
+              return (
+                <Form.Control isInvalid={doesAssetExist}>
+                  <Input size="md" css={styles.text}>
+                    <Input.Field
+                      {...props.field}
+                      placeholder="Search or paste custom address"
+                    />
+                  </Input>
+                  <Form.ErrorMessage>
+                    {`Please enter a valid address. This address has already been added.
+                    `}
+                  </Form.ErrorMessage>
+                </Form.Control>
+              );
+            }}
+          />
+        </Box.Stack>
+        <CardList isClickable={!editable}>
+          {/* TODO test that use token works */}
+          <>
+            {showUseTokenButton && (
+              <EthAssetCard
+                icon={
+                  <Avatar.Generated
+                    size={20}
+                    hash={
+                      (assetInfo?.address || '') + (assetInfo?.symbol || '')
+                    }
+                  />
+                }
+                name={assetInfo?.symbol || ''}
+                onAdd={() => {
+                  handlers.addAsset({
+                    asset: {
+                      address: assetInfo?.address,
+                      decimals: assetInfo?.decimals,
+                      symbol: assetInfo?.symbol,
+                    },
+                  });
+                }}
+              />
+            )}
+          </>
+          <>
+            {showCustomTokenButton && (
+              <EthAssetCard
+                icon={<Icon icon="HelpOctagon" />}
+                name={shortAddress(form.getValues('address'))}
+                onAdd={form.handleSubmit(onSubmitCustomToken)}
+              />
+            )}
+          </>
+          <>
+            {!(showCustomTokenButton || showUseTokenButton) &&
+              assets.map((asset, i) => (
+                <EthAssetCard
+                  key={`${asset.address || ''}${asset.symbol || ''}${String(
+                    i
+                  )}`}
+                  icon={
+                    asset.image ? (
+                      <Image alt=" " src={asset.image} />
+                    ) : (
+                      <Avatar.Generated
+                        size={20}
+                        hash={asset.address || asset.symbol || String(i)}
+                      />
+                    )
+                  }
+                  name={asset.symbol || ''}
+                  onPress={
+                    !editable
+                      ? () => {
+                          bridgeHandlers.changeAssetAddress({
+                            assetAddress: asset.address,
+                          });
+                        }
+                      : undefined
```

Comment:
this is only possible for conditionally rendering react code.  Since this code is js/ts && functions like [this](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Logical_AND)